platformio_esp32_envs.ini

```diff
@@ -133,3 +133,37 @@ build_flags               = ${env:test_ESP32_4M316k.build_flags} -DHAS_ETHERNET
 extends                   = env:test_ESP32-wrover-kit_4M316k
 platform                  = ${env:test_ESP32-wrover-kit_4M316k.platform}
 build_flags               = ${env:test_ESP32-wrover-kit_4M316k.build_flags} -DHAS_ETHERNET
+
+
+; A Lolin D32 PRO with 16MB Flash, allowing 4MB sketch size, and file storage using LittleFS filesystem
+[max_ESP32_16M]
+extends                   = esp32_common
+lib_ignore                = ESP8266WiFi, ESP8266Ping, ESP8266WebServer, ESP8266HTTPUpdateServer, ESP8266mDNS, ESPEasy_ESP8266Ping, ESP32_ping
+lib_deps                  = ${esp32_common.lib_deps}, https://github.com/lorol/LITTLEFS.git
+board_build.filesystem    = littlefs
+board_upload.maximum_size = 4194304
+build_flags               = ${esp32_common.build_flags}  
+                            -DFEATURE_ARDUINO_OTA
+                            -DUSE_LITTLEFS
```

Comment:
I've enabled LittleFS by default for this 16M Flash configuration, as SPIFFS was unreliable when configured > 1MB
8MB storage should probably only used when really needed, default should be the 2MB configuration.

Reply:
Hmm LittleFS still needs some patching in how the settings file is written.
Like I said before on LittleFS you can't update some portion somewhere in a file. All behind it has to be written again.
So saving may take quite a while (at least on ESP8266 when I tested it)

Reply:
Also the text in the drawing should be changed to LittleFS, when it is being used, as that tipped me off into making my comment :)

Reply:
> Like I said, for LittleFS we really should split the Config.dat file.
> Maybe a good idea for a new PR (when this is merged) to consider the Config.dat file as source and split it to several separate files (in a folder which will be hidden from the file system browser)
> When the Config.dat file is then requested, we can either stitch it together from the separate files, or first need to recreate it to access it via the file browser (or a fetch URL)

This is what you said before 👍